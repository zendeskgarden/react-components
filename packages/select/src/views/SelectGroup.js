/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import { retrieveTheme } from '@zendeskgarden/react-theming';
import { TextGroup } from '@zendeskgarden/react-textfields';

const COMPONENT_ID = 'select.select_group';

/**
 * Accepts all `<div>` props
 */
const SelectGroup = TextGroup.extend.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

SelectGroup.propTypes = {
  inline: PropTypes.bool
};

/** @component */
export default SelectGroup;
