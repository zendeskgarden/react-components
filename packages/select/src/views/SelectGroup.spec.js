import React from 'react';
import { shallow } from 'enzyme';
import SelectGroup from './SelectGroup';

describe('SelectGroup', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<SelectGroup />);

    expect(wrapper).toMatchSnapshot();
  });
});
