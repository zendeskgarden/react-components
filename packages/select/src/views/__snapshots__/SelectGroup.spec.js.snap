// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectGroup renders default styling 1`] = `
<div
  className="c-txt "
  data-garden-id="select.select_group"
  data-garden-version="version"
/>
`;
